### PR TITLE
Fabo/signup steps

### PR DESCRIPTION
**Pull request by faboweb**
Closes #ISSUE

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
@mariopino @colw do you have comments on the approach?

**Comment by jbibla** [comment]
very nice!

File: src/components/common/StepsForm.vue [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<template>
```
Comment (faboweb) [inline src/components/common/StepsForm.vue]: This is the important component

File: src/components/common/TmSessionSignUp.vue [hunk 2]
```diff
@@ -1,155 +1,142 @@
...
+      <div v-else>
+        <div v-if="!session.insecureMode" class="session-main">
+          <h2 class="session-title">
+            Create a new address
+          </h2>
+          <InsecureModeWarning />
+        </div>
+        <div v-else>
+          <div class="session-main">
+            <StepsForm :steps="steps" @done="success" ref="steps">
```
Comment (faboweb) [inline src/components/common/TmSessionSignUp.vue]: Here is an example implementation

File: src/components/common/TmSessionSignUp.vue [hunk 3]
```diff
@@ -1,155 +1,142 @@
...
+              <div slot="name">
```
Comment (faboweb) [inline src/components/common/TmSessionSignUp.vue]: Every step put into a slot with the name of the step

File: src/components/common/TmSessionSignUp.vue [hunk 4]
```diff
@@ -214,36 +210,48 @@ export default {
       }
     }
   },
+  created() {
+    this.steps = [
```
Comment (faboweb) [inline src/components/common/TmSessionSignUp.vue]: This defines the steps order and their slot names
It also defines the validation per step to do.

Comment (faboweb) [reply src/components/common/TmSessionSignUp.vue]: Nice thing is, this parent component still keeps the whole form and data in one place and can handle the whole data.

**Comment by mariopino** [comment]
Straightforward, I like it! We should change then create and recover flows to use the new component ... arg! :-)

**Comment by jbibla** [comment]
our new onboarding flow may not require so many steps so maybe we could close for now. does this still apply / would it be helpful for the action modals?

**Comment by faboweb** [comment]
won't there always be a "create account" and "import account" with at least 3 steps each?

**Comment by jbibla** [comment]
> won't there always be a "create account" and "import account" with at least 3 steps each?

probably. but since we're redesigning we might as well wait and see what we come up with.

**Comment by jbibla** [comment]
fabo is waiting for time on an airplane ✈️

**Pull request closed without merging**